src/lib/model/Model.ts

```diff
@@ -90,22 +97,53 @@ export function createModel<S extends object>(initialState: S): Model<S> {
     action: NormalAction<S, Arg, Data, E>
   ): NormalAccessorCreator<S, Arg, Data, E> {
     const prefix = prefixCounter++;
+    let timeoutId: number | undefined;
     const main = (arg: Arg) => {
       const hashArg = stableHash(arg);
       const key = `${prefix}/${hashArg}`;
-      const accessor = accessorRecord[key];
-      if (accessor) {
-        return accessor as NormalAccessor<S, Arg, Data, E>;
-      }
 
-      const newAccessor = new NormalAccessor(
+      const clearAccessorCache = () => {
+        const accessor = accessorRecord[key];
+        if (!accessor) return;
+        // Don't delete the accessor if it is mounted.
+        if (accessor.isMounted()) return;
+        delete accessorRecord[key];
+      };
+
+      const onMount = () => {
+        clearTimeout(timeoutId);
+      };
+
+      const onUnmount = () => {
+        timeoutId = window.setTimeout(clearAccessorCache, CLEAR_ACCESSOR_CACHE_TIME);
+      };
+
+      const constructorArgs: NormalConstructorArgs<S, Arg, Data, E> = {
         arg,
         action,
         updateState,
         getState,
-        subscribe,
-        notifyListeners
-      );
+        modelSubscribe: subscribe,
+        notifyModel: notifyListeners,
+        onMount,
+        onUnmount,
+      };
+
+      if (isServer()) {
+        // We don't need to cache the accessor in server side.
+        return new NormalAccessor(constructorArgs);
+      }
+
+      // Remove the clear cache timeout since the accessor is being used.
+      clearTimeout(timeoutId);
+      timeoutId = window.setTimeout(clearAccessorCache, CLEAR_ACCESSOR_CACHE_TIME);
```

Comment:
[dust] maybe `window.clearTimeout(timeoutId)` to keep the consistency?

Reply:
refactored in [bee2f3b](https://github.com/jason89521/daxus/pull/140/commits/bee2f3bf0513bf1d11e043d64b353baf1316c372)